src/lib/error/disabled-error.ts

```diff
@@ -1,9 +1,8 @@
-import BaseError from './base-error';
+import { UnleashError } from './api-error';
 
-class DisabledError extends BaseError {
+class DisabledError extends UnleashError {
     constructor(message: string) {
-        super(message, 422, 'DisabledError');
-        Error.captureStackTrace(this, this.constructor);
+        super({ message, name: 'DisabledError' });
```

Comment:
I've realized you're always passing `name: 'ClassName'` in every child error implementation. This is unnecessary and may lead to subtle typos. My suggestion is to stop passing name into super constructor and get it in UnleashError using this code as an inspiration:

```js
class ParentError extends Error {
    constructor(message) {
        super(message);
        console.log(
            `ParentError constructor called from ${this.constructor.name}`,
        );
    }
}

class ChildError extends ParentError {
    constructor(message) {
        super(message);
        console.log(
            `ChildError constructor called from ${this.constructor.name}`,
        );
    }
}

const error = new ChildError('This is an error message');
```

Reply:
I love this suggestion. Commit 35f3d4910a36e03d5ee6c6a45476918a5c88ff84:
- removes explicit class name passing, instead relying on `this.constructor.name`
- simplifies the base class constructor, allowing us to remove most of the constructors that we currently do have (based on an eslint feedback)

However, because the base constructor was protected, the compiler had some issues with building it. As such, I've made the constructor public for the base class. But I realized that JS does have abstract classes, which is helpful in this case. So I made the base class abstract (this seems to be the closest we can get to an interface while having it remain an Error type?).

Using constructor names instead of static strings does also mean that there's no such thing as exhaustive string matching for error types anymore, so I've had to relax switch statements and some types to allow general strings too. 

I've added an optional `name` parameter to the base error class to allow you to override the name that's set. This is primarily an escape hatch to use for the generic error class. It's just a suggestion and we can for sure change it.